CMSgov/onemac
# OY2-8634: refactor routing to improve extensibility

Story: https://qmacbis.atlassian.net/browse/OY2-8634
Endpoint: https://d3nhcfns9mr1pj.cloudfront.net/

To test:
1. Log in as a few types of user. Check that you can access what you should and that you can't access where you shouldn't. Also check that the header has the right links.
